R-package/R/lgb.plot.tree.R

```diff
@@ -0,0 +1,204 @@
+    modelDT[Node == 0, label := paste0("Tree ", Tree, "\n", label)]
+    modelDT[, shape := "rectangle"][Feature == "Leaf", shape := "oval"]
+    modelDT[, filledcolor := "Beige"][Feature == "Leaf", filledcolor := "Khaki"]
+    # in order to draw the first tree on top:
+    modelDT <- modelDT[order(-Tree)]
+    nodes <- DiagrammeR::create_node_df(
+        n         = nrow(modelDT)
+        , ID        = modelDT$ID
+        , label     = modelDT$label
+        , fillcolor = modelDT$filledcolor
+        , shape     = modelDT$shape
+        , data      = modelDT$Feature
+        , fontcolor = "black"
+    )
+    # round the edge labels to 4 s.f. if they are numeric
+    # as otherwise get too many decimal places and the diagram looks bad
+    # would rather not use suppressWarnings
+    numeric_idx <- suppressWarnings(!is.na(as.numeric(modelDT[["Split"]])))
+    modelDT[numeric_idx, Split := round(as.numeric(Split), 4)]
+    # replace indices with feature levels if rules supplied
+    
+    if (!is.null(rules)) {
+        for (f in names(rules)) {
+            modelDT[Feature == f & decision_type == "==", Split := .levels.to.names(Split, f, rules)]
+        }
+    }
+    # replace long split names with a message
+    modelDT[nchar(Split) > 500, Split := "Split too long to render"]
+    # create the edge labels
+    edges <- DiagrammeR::create_edge_df(
+        from = match(modelDT[Feature != "Leaf", c(ID)] %>% rep(2), modelDT$ID),
+        to = match(modelDT[Feature != "Leaf", c(Yes, No)], modelDT$ID),
+        label = modelDT[Feature != "Leaf", paste(decision_type, Split)] %>%
+            c(rep("", nrow(modelDT[Feature != "Leaf"]))),
+        style = modelDT[Feature != "Leaf", ifelse(Missing == Yes, "bold", "solid")] %>%
+            c(modelDT[Feature != "Leaf", ifelse(Missing == No, "bold", "solid")]),
+        rel = "leading_to"
+    )
+    # create the graph
+    graph <- DiagrammeR::create_graph(
+        nodes_df = nodes
+        , edges_df = edges
+        , attr_theme = NULL
+    )
+    graph <- DiagrammeR::add_global_graph_attrs(
+        graph = graph
+        , attr_type = "graph"
+        , attr = c("layout", "rankdir")
+        , value = c("dot", "LR")
+        )
+    graph <- DiagrammeR::add_global_graph_attrs(
+        graph = graph
+        , attr_type = "node"
+        , attr = c("color", "style", "fontname")
+        , value = c("DimGray", "filled", "Helvetica")
+    )
+    graph <- DiagrammeR::add_global_graph_attrs(
+        graph = graph
+        , attr_type = "edge"
+        , attr = c("color", "arrowsize", "arrowhead", "fontname")
+        , value = c("DimGray", "1.5", "vee", "Helvetica")
+    )
+    # render the graph
+    DiagrammeR::render_graph(graph)
+    return(invisible(NULL))
+}
+
+.zero_present <- function(x) {
+    sapply(strsplit(as.character(x), "||", fixed = TRUE), function(el) {
+        any(el == "0")
+    })
+    return(invisible(NULL))
+}
+
+.levels.to.names <- function(x, feature_name, rules) {
```

Comment:
```suggestion
.levels_to_names <- function(x, feature_name, rules) {
```

Please avoid using `.` in any of these private functions' names.